syntax = "proto3";

package control;

import "common/robot_types.proto";

service ControlService {
  rpc SetMotor(SetMotorRequest) returns (SetMotorResponse);

  rpc StopAll(StopAllRequest) returns (StopAllResponse);

  rpc LiveControl(stream LiveControlRequest)
      returns (stream LiveControlResponse);
}

message SetMotorRequest {
  common.MotorCommand command = 1;
}

message SetMotorResponse {
  bool accepted = 1;
}

message StopAllRequest {
}

message StopAllResponse {
  bool stopped = 1;
}

message LiveControlRequest {
  common.MotorCommand command = 1;
}

message LiveControlResponse {
  common.RobotState state = 1;
}
